src/Config/Factory.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PharIo\ComposerDistributor\Config;
+
+use PharIo\ComposerDistributor\File;
+use PharIo\ComposerDistributor\FileList;
+use PharIo\ComposerDistributor\Url;
+
+class Factory
+{
+	/**
+	 * Config json example:
+	 *
+	 * {
+	 *   "packageName": "phar-io/phive",
+	 *   "keyDirectory: "keys",
+	 *   "phars": [
+	 *     {
+	 *       "name": "phive",
+	 *       "file:" "https://github.com/phar-io/phive/releases/%version%/phive.phar",
+	 *       "signature:" "https://github.com/phar-io/phive/releases/%version%/phive.phar.asc"
+	 *     }
+	 *   ]
+	 * }
+	 *
+	 */
+	public static function createFromFile(string $configFile): Config
+	{
+		if (!is_file($configFile)) {
+			throw new \RuntimeException('Config file is missing');
+		}
+		$json = json_decode(file_get_contents($configFile), true);
+		if (json_last_error() !== JSON_ERROR_NONE) {
+			throw new \RuntimeException('Invalid json configuration');
+		}
+		$factory = new self();
+		return $factory->createConfig($json);
+	}
+
+	/**
+	 * Expected format:
+	 *
+	 * [
+	 *   'packageName'  => 'phar-io/phive',
+	 *   'keyDirectory' => 'keys',
+	 *   'phars'        => [
+	 *     [
+	 *       'name'      => 'phive',
+	 *       'file'      => 'https://github.com/phar-io/phive/releases/%version%/phive.phar',
+	 *       'signature' => 'https://github.com/phar-io/phive/releases/%version%/phive.phar.asc'
+	 *     ]
+	 *   ]
+	 * ]
+	 *
+	 */
+	private function createConfig(array $json): Config
+	{
+		$this->validateBaseConfig($json);
+		$this->validatePharsConfig($json);
+
+		return new Config($json['package'], $json['keyDir'], $this->createPhars($json));
+	}
+
+	private function createPhars(array $json): FileList
```

Comment:
Why don't we just pass in $json['phars'] here?

Reply:
Fixed